pkg/domain/infra/abi/images.go

```diff
@@ -351,65 +350,40 @@ func (ir *ImageEngine) Push(ctx context.Context, source string, destination stri
 	return pushError
 }
 
-// Transfer moves images from root to rootless storage so the user specified in the scp call can access and use the image modified by root
-func (ir *ImageEngine) Transfer(ctx context.Context, scpOpts entities.ImageScpOptions) error {
-	if scpOpts.User == "" {
+// Transfer moves images between root and rootless storage so the user specified in the scp call can access and use the image modified by root
+func (ir *ImageEngine) Transfer(ctx context.Context, source entities.ImageScpOptions, dest entities.ImageScpOptions, parentFlags []string) error {
+	if source.User == "" {
 		return errors.Wrapf(define.ErrInvalidArg, "you must define a user when transferring from root to rootless storage")
 	}
-	var u *user.User
-	scpOpts.User = strings.Split(scpOpts.User, ":")[0] // split in case provided with uid:gid
-	_, err := strconv.Atoi(scpOpts.User)
-	if err != nil {
-		u, err = user.Lookup(scpOpts.User)
-		if err != nil {
-			return err
-		}
-	} else {
-		u, err = user.LookupId(scpOpts.User)
-		if err != nil {
-			return err
-		}
-	}
-	uid, err := strconv.Atoi(u.Uid)
+	podman, err := os.Executable()
 	if err != nil {
 		return err
 	}
-	gid, err := strconv.Atoi(u.Gid)
+	// we cannot import entities into util, map the options
+	var sourceUtil util.ImageScpOptions
+	var destUtil util.ImageScpOptions
+	sourceMarshal, err := json.Marshal(source)
 	if err != nil {
 		return err
 	}
-	err = os.Chown(scpOpts.Save.Output, uid, gid) // chown the output because was created by root so we need to give th euser read access
+	err = json.Unmarshal(sourceMarshal, &sourceUtil)
 	if err != nil {
 		return err
 	}
 
-	podman, err := os.Executable()
+	destMarshal, err := json.Marshal(dest)
 	if err != nil {
 		return err
 	}
-	machinectl, err := exec.LookPath("machinectl")
+	err = json.Unmarshal(destMarshal, &destUtil)
```

Comment:
What is the difference between dest and destUtil?

Reply:
entities cannot be imported to utils. so we need to convert to the utils "friendly" struct before passing to that function.

Reply:
Can you move the function that is in utils to somewhere else, so you don't have to do this dance?